pkg/client/images.go

```diff
@@ -61,3 +62,10 @@ func (c *images) Delete(name string) (err error) {
 	err = c.r.Delete().Resource("images").Name(name).Do().Error()
 	return
 }
+
+// UpdateSignatures allows to modify signatures of given image.
+func (c *images) UpdateSignatures(image *imageapi.Image) (result *imageapi.Image, err error) {
```

Comment:
A signature is a single entity - trying to cram it into UpdateSignatures is
defeating the purpose of it being a unique item.

Named subresources are not allowed currently, and I don't think this
feature crosses the bar to introduce them.  So it has to be a top level
resource.

On Tue, Jun 14, 2016 at 8:59 AM, Michal Minar <notifications@github.com>
wrote:

> In pkg/client/images.go
> <https://github.com/openshift/origin/pull/9181#discussion_r66964209>:
>
> > @@ -61,3 +62,10 @@ func (c *images) Delete(name string) (err error) {
> >  	err = c.r.Delete().Resource("images").Name(name).Do().Error()
> >  	return
> >  }
> > +
> > +// UpdateSignatures allows to modify signatures of given image.
> > +func (c *images) UpdateSignatures(image *imageapi.Image) (result *imageapi.Image, err error) {
>
> We could also consider adding a constrained Name field to the signature
> that is settable by callers.
>
> If I get this correctly the *constraint* will be a name of containing
> image as a name's prefix. Followed by some separator (e.g. #). And the
> rest will be up to client's decision:
>
> signature.Name = "sha256:be29438d43e16dc0b63fbf64d7f989e85ccbab9546ac4b23ecd385194d0ff675#rhel7-base"
>
> I must say I'd still prefer the UpdateSignatures() call over adding some
> randomly generated (though constrained) name to signature. It seems more
> like an obstacle to API use.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9181/files/9ff4bf43548c758b6767b639b335681285fece48#r66964209>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p5HwMlS_HVh1hE3G-IPkv3jaowD-ks5qLqWUgaJpZM4IvEFx>
> .
>